### DISCUSSION TITLE
Help With Addon

### DISCUSSION THREAD

**Discussion by StickmanRed** [ScratchAddons/ScratchAddons#7399] [Help]
I'm trying to create an addon that "interrupts" [scratch-gui's sprite uploader](https://github.com/scratchfoundation/scratch-gui/blob/develop/src/lib/file-uploader.js#L211) by checking if the filetype is `image/gif` and creating a sprite (a `json` string) that contains the frames; similar to what the sprite uploader already does, but with the ability to preset certain values. I was thinking of modifying it directly, as a few sections of the `onion-skinning` addon overrode some prototype functions of `paper.project` ([example](https://github.com/ScratchAddons/ScratchAddons/blob/master/addons/onion-skinning/userscript.js#L48)). Can anything similar be done in this case, or should the addon be approached differently?

**Comment by WorldLanguages** [comment]
Also check [SVG uploads fix](https://github.com/ScratchAddons/ScratchAddons/blob/master/addons/fix-uploaded-svgs/userscript.js) and the [asset conflict resolution pull-request](https://github.com/ScratchAddons/ScratchAddons/pull/7232/files) as they're addons that interact with uploads and it might be useful to take a look at them.
